### ISSUE TITLE
[1.10+] Wands not taking proper amount of EXP per level  

### ISSUE THREAD

**Issue by SimPiko** [romelo333/notenoughwands#76]
- Modpack: [Faster Technology](https://minecraft.curseforge.com/projects/faster-technology)
 - Minecraft 1.10.2 (also happens on 1.12.2)
 - Forge 12.18.3.2239
 - NotEnoughWands 1.1x-1.5.4 (also happens on 1.12-1.6.1)

**Steps to reproduce:**
1. Get wand that uses Exp as fuel.
(depends on config, i've noticed it on Building Wand, Moving Wand and Displacement Wand)
2. Use it until you lose one level of Exp.

**Expected result:** Exp bar should be full (or almost full), no matter what Exp level you have
**Actual result:** [Exp bar shows weird values.](https://i.imgur.com/TCv6ADp.png)

While playing "Faster Technology" modpack, i've noticed that wands, when has taken whole Exp level, it restores only ~50% of EXP bar (at ~30lvl and up) and it restores more than a bar at ~5lvl or lower (the highlighted part of Exp bar goes waay to the right). I've quickly tested on mc1.12, it also happens there.

I've quickly looked in code here on GitHub, and looks like '[varia/Tools.java](https://github.com/romelo333/notenoughwands/blob/109722dfcd6ad8338d1706560f7109b0600f86e0/src/main/java/romelo333/notenoughwands/varia/Tools.java#L95)' uses pre-1.8 equations for calculating exp amount. It should be something like (at least according to [Minecraft Wiki](https://minecraft.gamepedia.com/Experience#Leveling_up)):

```
if (level > 0 && level < 17)
    return (int) (Math.pow(level, 2) + 6 * level);
else if (level > 16 && level < 32)
    return (int) (2.5 * Math.pow(level, 2) - 40.5 * level + 360);
else
    return (int) (4.5 * Math.pow(level, 2) - 162.5 * level + 2220);
```

Offtopic-wise, there are probably more mods that somehow forgot to update that [(i know Botania had that issue](https://github.com/Vazkii/Botania/issues/1931) and yes, i know i copy-pasted text from it), but that's another ~~story~~/issue.
Also, i don't know how to post issues properly and English isn't my first language, so i'm sorry if it's too wordy or weird too understand.

**Comment by RalphFox** [comment]
I'm guessing the issue I was about to report is related to this one. The amount of taken exp when you don't have enough for a single use on the current level on usage, it will take almost half of the previous level upon use, then back to taking a small amount.

Here is a gif:
http://g.recordit.co/iXFcCSIXqX.gif
![too much xp taken](http://g.recordit.co/iXFcCSIXqX.gif)

**Comment by RalphFox** [comment]
Actually it's worser. You can dupe xp with this.
I was level 56, close to 57, then I used the teleportation wand, and I jumped to level 57.
http://recordit.co/MsqldETdsX
http://g.recordit.co/MsqldETdsX.gif
